Run the full test despite failures

siddhesh opened the pull request:
Mark failures and continue execution of the testsuite.  Display a
summary at the end of each compiler run.

serge-sans-paille commented on Makefile:
```diff
@@ -26,14 +26,23 @@ STATIC_CHECK ?= false
 check:$(patsubst %,check-%,$(COMPILERS)) test_common.c
 
+$(RESULTFILE):
+	$(Q)cat /dev/null > $@
```
What an interesting idiom. I assume it's equivalent to `rm -f $@ && touch $@` ?

serge-sans-paille replied:
On a more general note, you're using `RESULTFILE` to accumulate the result of checks, which may interact surprisingly if some runs under `make -j`. May I suggest to use a result file per run and `grep` through all of them?

siddhesh replied:
> What an interesting idiom. I assume it's equivalent to `rm -f $@ && touch $@` ?

Kinda, the `cat` will retain the inode.

siddhesh replied:
> On a more general note, you're using `RESULTFILE` to accumulate the result of checks, which may interact surprisingly if some runs under `make -j`. May I suggest to use a result file per run and `grep` through all of them?

Good point; I'll split out the results and consolidate in the end.

serge-sans-paille commented on Makefile:
```diff
@@ -26,14 +26,23 @@ STATIC_CHECK ?= false
-	@echo "===== $$@ OK ====="
+check-$(1):$(RESULTFILE) $(patsubst %,run_%.$(1),$(TARGETS))
+	$(Q){ echo "===== $$@ DONE ====="; \
+	      echo -n "PASSED: "; grep OK $(RESULTFILE) | wc -l; \
```
`grep -c` is probably better than `grep | wc -l`

serge-sans-paille replied:
And `echo -n` => `printf "PASSED"` ?

siddhesh committed:
Run the full test despite failures

Mark failures and continue execution of the testsuite.  Display a
summary at the end of the run.

siddhesh commented:
Changes:

- Each test writes into its own intermediate test-result file
- At the end of each compiler run, a summary is put into a .result file
- At the end of the full test run, the .result files are dumped, showing the summary of the test run

serge-sans-paille commented:
LGTM, thanks!
